hackathon_site/review/test_admin.py

```diff
@@ -316,3 +335,137 @@ def test_shows_existing_review_fields(self):
+
+
+class AssignReviewerToTeamViewTestCase(SetupUserMixin, TestCase):
+    def setUp(self):
+        super().setUp()
+
+        self.view = reverse("admin:assign-reviewer-to-team")
+
+        self.user.is_staff = True
+        self.user.save()
+
+        self.user2_password = "abcdef123"
+        self.user2 = User.objects.create_user(
+            username="frederick@varley.com", password=self.user2_password
+        )
+        self.user2.is_staff = True
+        self.user2.save()
+
+        self.change_permissions = Permission.objects.filter(
+            Q(codename="view_application", content_type__app_label="registration")
+            | Q(codename="view_review", content_type__app_label="review")
+            | Q(codename="change_review", content_type__app_label="review"),
+        )
+
+        for perm in self.change_permissions:
+            self.user.user_permissions.add(perm)
+            self.user2.user_permissions.add(perm)
+
+        # team1 is entirely reviewed
+        self.team1 = self._make_full_registration_team(self_users=False)
+        for application in self.team1.applications.all():
+            self._review(application)
+
+        # team2 is missing a single review
+        self.team2 = self._make_full_registration_team(self_users=False)
+        for application in self.team2.applications.all()[:2]:
+            self._review(application)
+
+        # team3 is entirely unreviewed
+        self.team3 = self._make_full_registration_team(self_users=False)
+
+    def tearDown(self):
+        super().tearDown()
+        cache.clear()
+
+    def test_permission_denied_without_change_perm(self):
+        """
+        Without review.change_review permission, the view should not work
+        """
+        review_change_permission = Permission.objects.get(
+            codename="change_review", content_type__app_label="review"
+        )
+        self.user.user_permissions.remove(review_change_permission)
+        self._login()
+        response = self.client.get(self.view)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_takes_to_oldest_team_with_missing_reviews(self):
+        """
+        Test that the user is redirected to the oldest team with unreviewed
+        applications
+        """
+
+        self._login()
+        response = self.client.get(self.view)
+
+        # team2 should be the assigned view, since that one is the oldest
+        # team with missing reviews
+        expected_url = reverse(
+            "admin:review_teamreview_change", kwargs={"object_id": self.team2.id}
+        )
+        self.assertRedirects(response, expected_url)
+
+    def test_skips_team_assigned_to_another_user(self):
+        """
+        Test that the user is redirected to the oldest team with unreviewed
+        applications that is not assigned to another reviewed
+        """
+
+        # Login as self.user
+        self._login()
+        _ = self.client.get(self.view)
```

Comment:
Why do you assign this to anything - you don't use it anywhere else; just do `self.client.get(self.view)`

Reply:
No real reason, could have done it otherwise. _ is common for discarding variables, but it could cause issues if gettext_lazy is imported as _. Will remove.